src/dtgtk/expander.c

```diff
@@ -121,21 +121,24 @@ static gboolean _expander_scroll(GtkWidget *widget, GdkFrameClock *frame_clock,
 
   if(allocation.y < _start_pos.y)
   {
-    int offset = _start_pos.y - allocation.y - _start_pos.x + value;
+    const int offset = _start_pos.y - allocation.y - _start_pos.x + value;
     value -= offset;
     _start_pos.y = allocation.y;
   }
 
   float prop = 1.0f;
-  gboolean scroll_to_top = dt_conf_get_bool("darkroom/ui/scroll_to_module");
-  if((allocation.y + allocation.height - value > available.height && value < allocation.y) ||
-     (scroll_to_top && allocation.y != value))
+  const gboolean scroll_to_top = dt_conf_get_bool("darkroom/ui/scroll_to_module");
+  const int spare = available.height - allocation.height;
+  const int from_top = allocation.y - value;
+  const int move = MAX(from_top - scroll_to_top ? 0 : MAX(0, MIN(from_top, spare)),
```

Comment:
> Does it still work for you? Should I submit a new fix? After more testing? I'm just quite confused.

No it doesn't work for me either. I'm not sure I have tested this scenario though. I was looking at the fact that it should not scroll when I use the blend controls, but I've probably done that on a small module that does not extends much on the bottom.

Reply:
But does it "scroll to top" when expanding a module? (if you have that option selected in prefs) For me it basically doesn't scroll ever...

It's not that it surprises me that the code as is doesn't work. It is that I recall testing it extensively and (obviously) seeing it working, before submitting the PR. Maybe I was tired or something and made a last minute change, but I should have immediately caught this.

And I'm also surprised not to see any complaints that scroll-to-top stopped working (whereas there _are_ complaints that smoothly expanding is annoying).

Reply:
Seems I "simplified" from 
`  const int move = MAX(scroll_to_top ? from_top : from_top - MAX(0, MIN(from_top, spare)),
                       - MAX(0, spare - from_top));
`
at the last minute, without realising that's not the same thing (and doing sloppy testing). Apologies.

But now I also identified two more corner cases:
- when expanding a module below the currently open (and therefore now collapsing) module, I'm trying to keep it in the same place. But afterwards, the user can move it. If it then changes size, it again will be brought back to where it was when it was expanded. This should not happen.
- both left (lib) and right (iop) modules can be moving at the same time (for example because the history size changed). This can cause modules to move (upward) when they should be staying in the same place.

I'm working on solutions for all three.